pkg/fetch/nvd/api/cve/cve_test.go

```diff
@@ -1 +1,114 @@
 package cve_test
+
+import (
+	"fmt"
+	"io/fs"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"github.com/MaineK00n/vuls-data-update/pkg/fetch/nvd/api/cve"
+	"github.com/google/go-cmp/cmp"
+
+	"path/filepath"
+)
+
+func TestFetch(t *testing.T) {
+	tests := []struct {
+		name          string
+		apiKey        string
+		fixturePrefix string
+		hasError      bool
+	}{
+		{
+			name:          "empty",
+			fixturePrefix: "empty",
+		},
+		{
+			name:          "1 item",
+			fixturePrefix: "1_item",
+		},
+		{
+			name:          "Precisely single page",
+			fixturePrefix: "3_items",
+		},
+		{
+			name:          "Multiple pages",
+			fixturePrefix: "3_pages",
+		},
+		{
+			name:          "With API Key",
+			apiKey:        "foobar",
+			fixturePrefix: "3_pages",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				startIndex := "0"
+				if value := r.URL.Query().Get("startIndex"); value != "" {
+					startIndex = value
+				}
+				resultsPerPage := "2000"
+				if value := r.URL.Query().Get("resultsPerPage"); value != "" {
+					resultsPerPage = value
+				}
+
+				http.ServeFile(w, r, filepath.Join("testdata", "fixtures", fmt.Sprintf("%s-%s-%s.json", tt.fixturePrefix, startIndex, resultsPerPage)))
```

Comment:
How about dividing it into directories for each testcase?
```suggestion
				http.ServeFile(w, r, filepath.Join("testdata", "fixtures", fmt.Sprintf("%s/%s-%s.json", tt.fixturePrefix, startIndex, resultsPerPage)))
```